org.eclipse.swtchart.export/src/org/eclipse/swtchart/export/images/ImageSupplier.java

```diff
@@ -15,52 +15,76 @@
 import org.eclipse.swt.graphics.GC;
 import org.eclipse.swt.graphics.Image;
 import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.ImageDataProvider;
 import org.eclipse.swt.graphics.ImageLoader;
-import org.eclipse.swt.graphics.Rectangle;
-import org.eclipse.swt.graphics.Transform;
+import org.eclipse.swt.graphics.PaletteData;
+import org.eclipse.swt.graphics.Point;
 import org.eclipse.swtchart.extensions.core.BaseChart;
 
 public class ImageSupplier {
 
+	// Image data provider used locally to represent chart image
+	private class ChartImageDataProvider implements ImageDataProvider {
+
+		private ImageData imageData;
+
+		public ChartImageDataProvider(int width, int height) {
+
+			PaletteData palette = new PaletteData(0xFF, 0xFF00, 0xFF0000);
+			imageData = new ImageData(width, height, 32, palette);
+		}
+
+		@Override
+		public ImageData getImageData(int zoom) {
+
+			if(zoom != 100)
+				return null;
+			return imageData;
```

Comment:
I wonder if it won't be possible to just create a suitable imagedata object here for the given zoomlevel using height/width * (zoom / 100d) this might then also fixes your issue with the image beeing of different size than the chart.

Reply:
We are doing all this workaround because of bug :
https://bugs.eclipse.org/bugs/show_bug.cgi?id=558678

And now I'm facing another strange behavior with GC copy area : image is twice control size.
This is a call to **bitmapImageRepForCachingDisplayInRect** (native call) in GC.java line 458 that produce this size change. A workaround could be to scale image as I did in previously closed PR. But it's another workaround... 

I'm going to see if I can create a small snippet that produce this strange behavior and eventually create another bugzilla.

Please let me know what you think about that...

Reply:
I've not been able to reproduce this behavior with later version of SWT and I've finally decided to update my local swtchart target platform to Eclipse SDK 4.14, the original one from cloned repository was 4.7. With 4.14, this problem of image size disappears and the resulting code is even simpler as we no longer need any image data provider. I've not committed this last version waiting for your feedback. Meanwhile, have an happy end of year 😄

Reply:
I think you can just keep the code as is and we can see if we can update swtchart to use a more recent swt version.